tests/xharness/Logging/CaptureLog.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.IO;
+
+namespace Xharness.Logging {
+
+	// A log that captures data written to a separate file between two moments in time
+	// (between StartCapture and StopCapture).
+	public class CaptureLog : Log
+	{
+		public string CapturePath { get; private set; }
+		public string Path { get; private set; }
+
+		long startPosition;
+		long endPosition;
+		bool entire_file;
+		bool started;
+
+		public CaptureLog (ILogs logs, string path, string capture_path, bool entire_file = false)
+			: base (logs)
+		{
+			if (path == null)
+				throw new ArgumentNullException (nameof (path));
+
+			Path = path;
+			CapturePath = capture_path;
+			this.entire_file = entire_file;
+		}
+
+		public void StartCapture ()
+		{
+			if (entire_file)
+				return;
+
+			if (File.Exists (CapturePath))
+				startPosition = new FileInfo (CapturePath).Length;
+			started = true;
+		}
+
+		public void StopCapture ()
+		{
+			if (!started && !entire_file)
```

Comment:
I think it would be better to not store state of the class in two boolean variables but rather have a `CaptureState` enum with values `NotStarted`, `EntireFile` and `Started`. Because for example `started && entire_file` is not a valid state and the code should not let that happen.